### PR TITLE
Add note to testing suite instructions

### LINKED ISSUE #643: [Documentation]: Should remark that test data is required for running the tests.
### What would you like changed or added to the documentation and why?

This page says to run tests simply by pip installing and running `pytest`

https://neuroconv.readthedocs.io/en/main/developer_guide/testing_suite.html

However, this doesn't work since the test data needs to be downloaded. I only discovered that by scrolling down to the bottom of the page. I think this should be noted at the top.

### Do you have any interest in helping write or edit the documentation?

Yes.

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/catalystneuro/neuroconv/blob/main/.github/CODE_OF_CONDUCT.rst)
- [X] Have you ensured this bug was not already [reported](https://github.com/catalystneuro/neuroconv/issues)?

### PR DESCRIPTION
**Pull request by CodyCBakerPhD**
fix #643

### CODE REVIEW

File: docs/developer_guide/testing_suite.rst [hunk 1]
```diff
@@ -29,6 +29,10 @@ Then simply run all tests with pytest
 
   pytest
 
+.. note::
+
+  You will likely observe many failed tests at this point, if you have not configured the other aspects of the testing suite shown below.
```
Comment (h-mayorquin) [inline docs/developer_guide/testing_suite.rst:34]:
```suggestion
  You will likely observe many failed tests if the test data is not available. See the section 'Testing on Example Data' for instructions on how to download the test data.
```

Comment (CodyCBakerPhD) [reply docs/developer_guide/testing_suite.rst:34]: I kept it more general because you'll also see failures if you haven't done `neuroconv[full]`, right?

Comment (CodyCBakerPhD) [reply docs/developer_guide/testing_suite.rst:34]: Oh, but in the first section we do specify that. OK

**Review by h-mayorquin — APPROVED** [review]
Thanks for addressing this so quicklly @CodyCBakerPhD . 

Hope this informative in the future @magland .

**Comment by magland** [comment]
Thanks @CodyCBakerPhD and @h-mayorquin !

**Pull request merged**